src/core/thread/mle.hpp

```diff
@@ -1775,6 +1686,16 @@ class Mle : public InstanceLocator
         kDataRequestActive, ///< Data Request has been sent, Data Response is expected.
     };
 
+    struct Metadata // for delayed response
```

Comment:
How about something more descriptive, like `ResponseInfo`?

Reply:
Sounds good. I changed it to `DelayedResponseMetadata`.